implementations/rust/ockam/ockam_transport_ble/src/driver/mutex.rs

```diff
@@ -0,0 +1,272 @@
+//! A naive, single-threaded Async Mutex implementation
+//!
+//! It is used by the embedded implementation and is only safe on
```

Comment:
It would be nice to find a way to enforce this somewhere, to prevent accidental misuse. But we have this as a broader issue.

Reply:
Amen! :-P